### PR TITLE
Add reconciler nightly job for cluster

### PR DESCRIPTION
**Pull request by sayanh**
**Description**

Changes proposed in this pull request:

- Add reconciler nightly job for cluster creation and deletion which will be in turn used by the reconciler nightly e2e job to periodically test reconciliation flow with fast integration test. 

**Related issue(s)**

https://github.com/kyma-incubator/reconciler/issues/75

### CODE REVIEW

**Review by suleymanakbas91 — APPROVED** [review]

File: prow/jobs/incubator/reconciler/reconciler.yaml [hunk 1]
```diff
@@ -230,4 +230,70 @@ periodics: # runs on schedule
...
+          path_alias: github.com/kyma-project/test-infra
+          base_ref: main
+      reporter_config:
+        slack:
+          channel: kyma-prow-dev-null
+      spec:
+        containers:
+          - image: "eu.gcr.io/kyma-project/test-infra/kyma-integration:v20210902-035ae0cc-k8s1.18"
+            securityContext:
+              privileged: true
```
Comment (dekiel) [inline prow/jobs/incubator/reconciler/reconciler.yaml:274]: Do you need this level of privileges? If not you can override this and set to false in jobconfig section.

File: prow/scripts/cluster-integration/reconciler-gardener-long-lasting.sh [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+# - GARDENER_ZONES - Gardener compute zones inside the region
+# - GARDENER_CLUSTER_VERSION - Version of the Kubernetes cluster
+# - GARDENER_KYMA_PROW_KUBECONFIG - Kubeconfig of the Gardener service account
+# - GARDENER_KYMA_PROW_PROJECT_NAME - Name of the gardener project where the cluster will be integrated.
+# - GARDENER_KYMA_PROW_PROVIDER_SECRET_NAME - Name of the secret configured in the gardener project to access the cloud provider
+# - MACHINE_TYPE - (optional) machine type
+#
+#Please look in each provider script for provider specific requirements
+
+function delete_cluster_if_exists(){
```
Comment (dekiel) [inline prow/scripts/cluster-integration/reconciler-gardener-long-lasting.sh:21]: Did you check if you can use functions from lib directory? **https://github.com/kyma-project/test-infra/tree/main/prow/scripts/lib/gardener**

We have functions for creating, removing and connecting to shoot clusters.

**Review by dekiel — CHANGES_REQUESTED** [review]

File: prow/jobs/incubator/reconciler/reconciler.yaml (hunk 1 above)
Comment (sayanh) [reply prow/jobs/incubator/reconciler/reconciler.yaml:274]: Changed to false.

Comment (sayanh) [reply prow/jobs/incubator/reconciler/reconciler.yaml]: Fixed

File: prow/scripts/cluster-integration/reconciler-gardener-long-lasting.sh (hunk 2 above)
Comment (sayanh) [reply prow/scripts/cluster-integration/reconciler-gardener-long-lasting.sh:21]: The creation of shoot clusters in the library is different from my need. Hence I’d stick to my approach. The deletion can be reused but with a check, if exists. I reused `gardener::deprovision_cluster` for now. Connecting to shoot is missing completely.

**Review by dekiel — APPROVED** [review]

**Review by dekiel — APPROVED** [review]

**Pull request merged**